examples/01_plotting/plot_carpet.py

```diff
@@ -0,0 +1,43 @@
+"""
+Visualizing global patterns with a carpet plot
+==============================================
+
+A common quality control step for functional MRI data is to visualize the data
+over time in a carpet plot (also known as a Power plot or a grayplot).
+
+The :func:`nilearn.plotting.plot_carpet()` function generates a carpet plot
+from a 4D functional image.
+"""
+
+###############################################################################
+# Fetching data from ADHD dataset
+# -------------------------------
+from nilearn import datasets
+
+adhd_dataset = datasets.fetch_adhd(n_subjects=1)
+
+# Print basic information on the dataset
+print('First subject functional nifti image (4D) is at: %s' %
+      adhd_dataset.func[0])  # 4D data
+
+###############################################################################
+# Deriving a mask
+# ---------------
+from nilearn import masking
+
+# Build an EPI-based mask because we have no anatomical data
+mask_img = masking.compute_epi_mask(adhd_dataset.func[0])
+
+###############################################################################
+# Visualizing global patterns over time
+# -------------------------------------
+import matplotlib.pyplot as plt
+
+from nilearn.plotting import plot_carpet
+
+fig, ax = plt.subplots(figsize=(10, 5))
+
+fig = plot_carpet(adhd_dataset.func[0], mask_img, figure=fig, axes=ax)
+
+fig.tight_layout()
```

Comment:
Ideally, I think that we should try to work on the defaults of plot_carpet so that this example looks fine without creating the figure and calling tight_layout. Maybe a figure size computed as a function of the data (with min and max limits to avoid pathological situations)?

Reply:
The axis labels are shifted out to give it whitespace between the axes and the data. I don't know of a way to account for that at the point where the fig object is created, so I think either the labels have to be up against the data (which doesn't look as good) or tight_layout needs to be used in the function.